dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -215,6 +216,33 @@ public void checkEntryPermissions(final Optional<User> user, final Entry<? exten
         }
     }
 
+    @GET
+    @UnitOfWork
+    @Path("/{entryId}/versions/{versionId}/descriptionMetrics")
+    @ApiOperation(value = "Retrieve metrics on the description of an entry")
+    @Operation(operationId = "getDescriptionMetrics", description = "Retrieve metrics on the description of an entry", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiResponse(responseCode = HttpStatus.SC_OK + "", description = "Successfully calculated description metrics", content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = DescriptionMetrics.class)))
+    public DescriptionMetrics calculateDescriptionMetrics(@Parameter(hidden = true, name = "user")@Auth Optional<User> user,
+        @Parameter(name = "entryId", description = "Entry to retrieve the version from", required = true, in = ParameterIn.PATH) @PathParam("entryId") Long entryId,
+        @Parameter(name = "versionId", description = "Version to retrieve the sourcefile types from", required = true, in = ParameterIn.PATH) @PathParam("versionId") Long versionId) {
+        Entry<? extends Entry, ? extends Version> entry = toolDAO.getGenericEntryById(entryId);
+        checkEntry(entry);
+
+        checkEntryPermissions(user, entry);
+
+        Version version = versionDAO.findVersionInEntry(entryId, versionId);
+        if (version == null) {
+            throw new CustomWebApplicationException("Version " + versionId + " does not exist for this entry", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        String description = version.getVersionMetadata().getDescription();
+        if (description == null) {
+            throw new CustomWebApplicationException("Version " + versionId + " does not have a description", HttpStatus.SC_BAD_REQUEST);
```

Comment:
You could argue this both ways, but I would lean towards this returning a DescriptionMetrics whose values all 0

Reply:
Removed the null check, since DescriptionMetrics already handles them it'll return 0s for all metrics.